enhancements/cluster-logging/loki-tokenized-auth-enablement.md

```diff
@@ -0,0 +1,407 @@
+
+```yaml
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: logging-loki
+  namespace: openshift-logging
+spec:
+  cloudTokenPath: /var/run/secrets/storage/serviceaccount/token
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AzureProviderSpec
+    AzureClientID: clientid
+    AzureTenantID: tenantid
+    AzureRegion: centralus
+    AzureSubscriptionID: subscriptionid
+  secretRef:
+    name: logging-loki-managed-credentials
+    namespace: openshift-logging
+  serviceAccountNames:
+  - logging-loki
+  - logging-loki-ruler
+```
+
+#### GCP WIF Support
+
+GCP WIF support will be supported only in the form of the upstream support as per CCO missing respectable bits.
+
+### API Extensions
+
+The present proposal introduces a new optional field in the `ObjectStorageSecretSpec` namely `CredentialsMode`. If the user does not provide a value the `CredentialsMode` is automatically detected either from the provided secret fields or the operator environment variables (latter applies only in OpenShift cluters.). The selected or detected `CredentialsMode` is populated in addition in the `status.storage.credentialMode` field.
+
+```go
+/ CredentialMode represents the type of authentication used for accessing the object storage.
+//
+// +kubebuilder:validation:Enum=static;token;managed
+type CredentialMode string
+
+const (
+    // CredentialModeStatic represents the usage of static, long-lived credentials stored in a Secret.
+    // This is the default authentication mode and available for all supported object storage types.
+    CredentialModeStatic CredentialMode = "static"
+    // CredentialModeToken represents the usage of short-lived tokens retrieved from a credential source.
+    // In this mode the static configuration does not contain credentials needed for the object storage.
+    // Instead, they are generated during runtime using a service, which allows for shorter-lived credentials and
+    // much more granular control. This authentication mode is not supported for all object storage types.
+    CredentialModeToken CredentialMode = "token"
+    // CredentialModeManaged represents the usage of short-lived tokens retrieved from a credential source.
+    // This mode is similar to CredentialModeToken,but instead of having a user-configured credential source,
+    // it is configured by the environment, for example the Cloud Credential Operator in OpenShift.
+    // This mode is only supported for certain object storage types in certain runtime environments.
+    CredentialModeManaged CredentialMode = "managed"
```

Comment:
```suggestion
    // CredentialModeTokenCCO represents the usage of short-lived tokens retrieved from a credential source.
    // This mode is similar to CredentialModeToken, but instead of having a user-configured credential source,
    // it is configured by the environment and the operator relies on the Cloud Credential Operator to provide
    // a secret. This mode is only supported for certain object storage types in certain runtime environments.
    CredentialModeTokenCCO CredentialMode = "token-cco"
```